modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/gc/io/GcIo.java

```diff
@@ -118,15 +131,16 @@ default int compare(long pageAddr, int idx, GarbageCollectionRowVersion rowVersi
      * @param idx Element's index.
      * @param partitionId Partition id to enrich read partitionless links.
      */
-    default GarbageCollectionRowVersion getRow(long pageAddr, int idx, int partitionId) {
+    default GcRowVersion getRow(long pageAddr, int idx, int partitionId) {
         int offset = offset(idx);
 
         long rowIdMsb = getLong(pageAddr, offset + ROW_ID_MSB_OFFSET);
         long rowIdLsb = getLong(pageAddr, offset + ROW_ID_LSB_OFFSET);
 
-        return new GarbageCollectionRowVersion(
+        return new GcRowVersion(
                 new RowId(partitionId, rowIdMsb, rowIdLsb),
-                HybridTimestamps.readTimestamp(pageAddr, offset + ROW_TIMESTAMP_OFFSET)
+                requireNonNull(HybridTimestamps.readTimestamp(pageAddr, offset + ROW_TIMESTAMP_OFFSET)),
```

Comment:
What's up with the `requireNonNull`? We can validate it during insertions, for example, not during the read

Reply:
I think that this is not necessary, I did it on a question from Alexander. I'll delete this.